File: A2/Sem3/MAP/Laborator/L3/src/domain/validation/Validator.java
package domain.validation;
import exceptii.ValidationException;

public interface Validator<T> {
    /**
     * Valideaza un obiect
     * @param entity - obiectul de validat
     * @throws ValidationException - daca obiectul nu este valid
     */
    void validate(T entity) throws ValidationException;
}

File: A1/Semestrul 1/Examen/Validator/validator.py
from Domain.carte import Carte
from exceptii import ValidationException

class Validator:
    """
    clasa care valideaza datele unei carti
    """
    @staticmethod
    def valideaza(carte: Carte):
        """
        :param carte: carte
        :raise: ValidationException daca unul dintre campuri e invalid
        """
        if carte.id < 0:
            raise ValidationException("Id invalid!")
        if carte.titlu == "":
            raise ValidationException("Titlu invalid!")
        if carte.autor == "":
            raise ValidationException("Autor invalid")
        if carte.anAp < 0:
            raise ValidationException("Anul aparitiei invalid!")
